sample_project/Assets/SampleViewer/Samples/FeatureLayer/Scripts/ArcGISFeatureLayerComponent.cs

```diff
@@ -0,0 +1,142 @@
+using Esri.ArcGISMapsSDK.Components;
+using System;
+
+public class ArcGISFeatureLayerComponent : MonoBehaviour
+{
+    [System.Serializable]
+    public struct QueryLink
+    {
+        public string Link;
+        public string[] RequestHeaders;
+    }
+
+    [System.Serializable]
+    public class GeometryData
+    {
+        public List<double> coordinates = new List<double>();
+    }
+
+    [System.Serializable]
+    public class PropertyData
+    {
+        public List<string> propertyNames = new List<string>();
+        public List<string> data = new List<string>();
+    }
+
+    [System.Serializable]
+    public class FeatureQueryData
+    {
+        public GeometryData geometry = new GeometryData();
+        public PropertyData properties = new PropertyData();
+    }
+
+    private List<FeatureQueryData> Features = new List<FeatureQueryData>();
+    private FeatureData featureInfo;
+    public List<GameObject> FeatureItems = new List<GameObject>();
+    [SerializeField] private GameObject featurePrefab;
+    private float spawnHeight = 10000.0f;
+
+    public JToken[] jFeatures;
+    public QueryLink webLink;
+
+    public void CreateLink(string link)
+    {
+        if (link != null)
+        {
+            foreach (var header in webLink.RequestHeaders)
+            {
+                if (!link.ToLower().Contains(header))
+                {
+                    link += header;
+                }
+            }
+
+            webLink.Link = link;
+        }
+    }
+
+    public IEnumerator GetFeatures()
+    {
+        // To learn more about the Feature Layer rest API and all the things that are possible checkout
+        // https://developers.arcgis.com/rest/services-reference/enterprise/query-feature-service-layer-.htm
+
+        UnityWebRequest Request = UnityWebRequest.Get(webLink.Link);
+        yield return Request.SendWebRequest();
+
+        if (Request.result != UnityWebRequest.Result.Success)
+        {
+            Debug.Log(Request.error);
+        }
+        else
+        {
+            CreateGameObjectsFromResponse(Request.downloadHandler.text);
+        }
+    }
+
+    private void CreateGameObjectsFromResponse(string response)
+    {
+        // Deserialize the JSON response from the query.
+        var jObject = JObject.Parse(response);
+        jFeatures = jObject.SelectToken("features").ToArray();
+
+        if (jFeatures[0].SelectToken("geometry").SelectToken("type").ToString().ToLower() != "point")
+        {
+            return;
+        }
+
+        CreateFeatures();
+    }
+
+    private void CreateFeatures()
+    {
+        foreach (var feature in jFeatures)
+        {
+            var currentFeature = new FeatureQueryData();
+            var featureItem = Instantiate(featurePrefab, this.transform);
+            //Layer 7 because that is the index of the layer created specifically for feature layers so that they ignore themselves for raycasting.
+            featureItem.layer = 7;
```

Comment:
Should we make this a serialized/public property to make is used more generally? (similar to link and headers and prefab)